src/openlcb/DccAccyProducer.cxxtest

```diff
@@ -0,0 +1,274 @@
+#include "utils/async_if_test_helper.hxx"
+
+#include "DccAccyProducer.hxx"
+
+namespace openlcb
+{
+
+using namespace std::placeholders;
+
+class DccAccyProducerTest : public AsyncNodeTest
+{
+public:
+    MOCK_METHOD2(state_callback1, void(unsigned, bool));
+    MOCK_METHOD2(state_callback2, void(unsigned, bool));
+
+protected:
+    DccAccyProducerTest()
+    {
+        print_all_packets();
+        wait();
+    }
+
+public:
+    SyncNotifiable n_;
+
+    class ImplFlow : public StateFlowBase
+    {
+    public:
+        ImplFlow(DccAccyProducerTest *test)
+            : StateFlowBase(test->ifCan_.get())
+            , dap1_(test->node_,
+                 std::bind(&DccAccyProducerTest::state_callback1, test, _1, _2))
+            , dap2_(test->node_,
+                 std::bind(&DccAccyProducerTest::state_callback2, test, _1, _2))
+            , test_(test)
+        {
+        }
+
+        ~ImplFlow()
+        {
+            auto *b1 = full_allocation_result(&dap1_);
+            auto *b2 = full_allocation_result(&dap2_);
+            if (b1)
+            {
+                b1->unref();
+            }
+            if (b2)
+            {
+                b2->unref();
+            }
+        }
+
+        DccAccyProducerInput *response1()
```

Comment:
did you know that response1 and response2 will always return the same data?

full_allocation_result only takes the argument as a type parameter, and always returns allocationResult_ cast to that type, which only exists once.